pkg/sourcetransformer/service_test.go

```diff
@@ -83,42 +143,120 @@ func TestService_sourceTransformFn(t *testing.T) {
 			}),
 			args: args{
 				ctx: context.Background(),
-				d: &stpb.SourceTransformRequest{
-					Keys:      []string{"client"},
-					Value:     []byte(`test`),
-					EventTime: timestamppb.New(time.Time{}),
-					Watermark: timestamppb.New(time.Time{}),
+				d: &proto.SourceTransformRequest{
+					Request: &proto.SourceTransformRequest_Request{
+						Keys:      []string{"client"},
+						Value:     []byte(`test`),
+						EventTime: timestamppb.New(time.Time{}),
+						Watermark: timestamppb.New(time.Time{}),
+					},
 				},
 			},
-			want: &stpb.SourceTransformResponse{
-				Results: []*stpb.SourceTransformResponse_Result{
+			want: &proto.SourceTransformResponse{
+				Results: []*proto.SourceTransformResponse_Result{
 					{
 						EventTime: timestamppb.New(testTime),
 						Tags:      []string{DROP},
-						Value:     []byte{},
+						Value:     nil,
 					},
 				},
 			},
-			wantErr: false,
 		},
 	}
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			fs := &Service{
+			svc := &Service{
 				Transformer: tt.handler,
 			}
-			// here's a trick for testing:
-			// because we are not using gRPC, we directly set a new incoming ctx
-			// instead of the regular outgoing context in the real gRPC connection.
-			ctx := context.Background()
-			got, err := fs.SourceTransformFn(ctx, tt.args.d)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("SourceTransformFn() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("SourceTransformFn() got = %v, want %v", got, tt.want)
-			}
+
+			conn := newServer(t, func(server *grpc.Server) {
+				proto.RegisterSourceTransformServer(server, svc)
+			})
+
+			client := proto.NewSourceTransformClient(conn)
+			stream, err := client.SourceTransformFn(context.Background())
+			assert.NoError(t, err, "Creating stream")
+
+			doHandshake(t, stream)
+
+			err = stream.Send(tt.args.d)
+			assert.NoError(t, err, "Sending message over the stream")
+
+			got, err := stream.Recv()
+			assert.NoError(t, err, "Receiving message from the stream")
+
+			assert.Equal(t, got.Results, tt.want.Results)
 		})
 	}
 }
+
+func doHandshake(t *testing.T, stream proto.SourceTransform_SourceTransformFnClient) {
```

Comment:
nit: please follow the format to name the unit test. e.g., `TestService_sourceTransformFn`

Reply:
This is a helper function, not a top-level test function.